### PR TITLE
Site Editor: Add View Link

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632 

## What?

This PR continues the work on the great unification between post and site editors. This PR moves the ViewLink component from the edit-post package to the editor package and reuse the same panel in the site editor as well. This means that for post types that are viewable in the frontend, there will be a link to "view the page" in the frontend in the header of the editor in the site editor, similar to the post editor.

## Testing instructions

- Open a published page in the site editor, check that you can see a "view" link in the header next to the device previews.

### CODE REVIEW

File: packages/edit-post/src/components/header/index.js [hunk 1]
```diff
@@ -189,7 +188,7 @@ function Header( {
 					className="edit-post-header__post-preview-button"
 					forceIsAutosaveable={ hasActiveMetaboxes }
 				/>
-				<ViewLink />
+				<PostViewLink />
```
Comment (t-hamano) [inline packages/edit-post/src/components/header/index.js]:
```suggestion
				<PostViewLink showIconLabels={ showIconLabels } />
```

It looks like the `showIconLabels` prop is not being passed. Without this, even if "Show button text labels" is enabled, a tooltip with the same text will be displayed when the button text is focused.

File: packages/edit-site/src/components/header-edit-mode/index.js [hunk 2]
```diff
@@ -217,6 +217,7 @@ export default function HeaderEditMode( { setListViewToggleElement } ) {
 							/>
 						</div>
 					) }
+					<PostViewLink />
```
Comment (t-hamano) [inline packages/edit-site/src/components/header-edit-mode/index.js]:
```suggestion
					<PostViewLink showIconLabels={ showIconLabels } />
```

**Review by t-hamano — COMMENTED** [review]
When "Show button text labels" is enabled, when I open any page in the Site Editor in edit mode and reload the browser, the button text will switch from "View Post" to "View Page". Is there any way to get around this?

https://github.com/WordPress/gutenberg/assets/54422211/7dd508a7-077c-4c52-b386-9a74215c5f55

File: packages/edit-post/src/components/header/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-post/src/components/header/index.js]: oh yeah, forgot to pass this prop. Thanks for noticing.

**Comment by t-hamano** [comment]
> When "Show button text labels" is enabled, when I open any page in the Site Editor in edit mode and reload the browser, the button text will switch from "View Post" to "View Page". Is there any way to get around this?

Maybe I don't understand exactly, but does it make sense to refer to `hasFinishedResolution` and wait for it to be resolved?

**Review by ntsekouras — APPROVED** [review]
Looks good, thank you! I'm approving since the issue @t-hamano mentions already exists in trunk and we can update easily.

If we really want to defer here we could just use:
`label={ label }`

The `view_item` label already default is ‘View Post’ / ‘View Page’.

**Pull request merged**